components/gitpod-db/src/typeorm/migration/1651188368768-VolumeSnapshotCreation.ts

```diff
@@ -0,0 +1,22 @@
+/**
+ * Copyright (c) 2022 Gitpod GmbH. All rights reserved.
+ * Licensed under the GNU Affero General Public License (AGPL).
+ * See License-AGPL.txt in the project root for license information.
+ */
+
+import { MigrationInterface, QueryRunner } from "typeorm";
+import { tableExists } from "./helper/helper";
+
+export class VolumeSnapshotCreation1651188368768 implements MigrationInterface {
+    public async up(queryRunner: QueryRunner): Promise<void> {
+        await queryRunner.query(
+            `CREATE TABLE IF NOT EXISTS d_b_volume_snapshot (  id char(36) NOT NULL, creationTime timestamp(6) NOT NULL DEFAULT CURRENT_TIMESTAMP(6) ON UPDATE CURRENT_TIMESTAMP(6),  originalWorkspaceId char(36) NOT NULL, volumeHandle varchar(255) NOT NULL, PRIMARY KEY (id),  KEY ind_originalWorkspaceId (originalWorkspaceId),  KEY ind_dbsync (creationTime)) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4;`,
+        );
+    }
+
+    public async down(queryRunner: QueryRunner): Promise<void> {
```

Comment:
Do you really want to drop the table in case we roll-back? I think here it's sensible to not drop it and keep it around, just unused. It's risky in the sense that server is never just a single replica, so we can't guarantee we won't be running code which will try to access the table and another which won't. Normally, we'd first create the table without any usage and then in later PRs, we'd start using it, such that it doesn't get dropped if the later PR needs to be reverted.

Reply:
Good question. I was following other migration examples, and thought that was the way to do it. :shrug:

Reply:
@easyCZ updated. PTAL :pray: